### PR TITLE
ci: add mergify rules for release-3.5 backport

### PR DESCRIPTION
**Pull request by humblec**
This add automatic backport rules for version 3.5

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -38,6 +38,24 @@ queue_rules:
...
+              - "status-success=lint-extras"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.21"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.22"
+              - "status-success=ci/centos/mini-e2e-helm/k8s-1.23"
+              - "status-success=ci/centos/mini-e2e/k8s-1.21"
+              - "status-success=ci/centos/mini-e2e/k8s-1.22"
+              - "status-success=ci/centos/mini-e2e/k8s-1.23"
+              - "status-success=ci/centos/upgrade-tests-cephfs"
+              - "status-success=ci/centos/upgrade-tests-rbd"
+              - "status-success=DCO"
```
Comment (Madhu-1) [inline .mergify.yml]: why this is needed isn't it already converted in the above rule?

File: .mergify.yml [hunk 2]
```diff
@@ -175,6 +193,44 @@ pull_request_rules:
         name: default
       dismiss_reviews: {}
       delete_head_branch: {}
+  - name: backport patches to release-v3.5 branch
+    conditions:
+      - base=devel
+      - label=backport-to-release-v3.5
```
Comment (Madhu-1) [inline .mergify.yml:181]: please create a label for this one

File: .mergify.yml [hunk 3]
```diff
@@ -175,6 +193,44 @@ pull_request_rules:
...
+      - "approved-reviews-by=@ceph/ceph-csi-maintainers"
+      - "status-success=codespell"
+      - "status-success=multi-arch-build"
+      - "status-success=go-test"
+      - "status-success=commitlint"
+      - "status-success=golangci-lint"
+      - "status-success=mod-check"
+      - "status-success=lint-extras"
+      - "#changes-requested-reviews-by=0"
+      - "status-success=ci/centos/mini-e2e-helm/k8s-1.21"
```
Comment (Madhu-1) [inline .mergify.yml:203]: why external-storage tests are not considered here?

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: .mergify.yml (hunk 1 above)
Comment (Rakshith-R) [reply .mergify.yml]: > why this is needed isn't it already converted in the above rule?

if we change rules for devel branch in the future, we will need to create the above set of rules again ?

File: .mergify.yml [hunk 4]
```diff
@@ -38,6 +38,24 @@ queue_rules:
...
               - "status-success=DCO"
+          - and:
+              - base=release-v3.5
+              - "status-success=codespell"
+              - "status-success=multi-arch-build"
+              - "status-success=go-test"
+              - "status-success=commitlint"
+              - "status-success=golangci-lint"
+              - "status-success=mod-check"
+              - "status-success=lint-extras"
```
Comment (Rakshith-R) [inline .mergify.yml]:
```suggestion
              - "status-success=lint-extras"
              - "status-success=ci/centos/k8s-e2e-external-storage/1.21"
              - "status-success=ci/centos/k8s-e2e-external-storage/1.22"
              - "status-success=ci/centos/k8s-e2e-external-storage/1.23"
```

File: .mergify.yml [hunk 5]
```diff
@@ -175,6 +193,44 @@ pull_request_rules:
...
+      - "approved-reviews-by=@ceph/ceph-csi-contributors"
+      - "approved-reviews-by=@ceph/ceph-csi-maintainers"
+      - "status-success=codespell"
+      - "status-success=multi-arch-build"
+      - "status-success=go-test"
+      - "status-success=commitlint"
+      - "status-success=golangci-lint"
+      - "status-success=mod-check"
+      - "status-success=lint-extras"
+      - "#changes-requested-reviews-by=0"
```
Comment (Rakshith-R) [inline .mergify.yml]:
```suggestion
      - "#changes-requested-reviews-by=0"
      - "status-success=ci/centos/k8s-e2e-external-storage/1.21"
      - "status-success=ci/centos/k8s-e2e-external-storage/1.22"
      - "status-success=ci/centos/k8s-e2e-external-storage/1.23"
```

**Review by Rakshith-R — CHANGES_REQUESTED** [review]

File: .mergify.yml (hunk 1 above)
Comment (Madhu-1) [reply .mergify.yml]: yes, change it when required?

Comment (humblec) [reply .mergify.yml]: it is better to keep seperate rules for maintained branches, isnt it ? and keep devel intact @Madhu-1 ?

Comment (humblec) [reply .mergify.yml]: I am fine with either way, if we dont need release specific rules , I can remove existing `release-3.4` set too. wdyt @Madhu-1 @Rakshith-R

Comment (Madhu-1) [reply .mergify.yml]: release-3.4 is too old and new tests are added in devel branch so you cannot remove it. am fine with this one.

File: .mergify.yml [hunk 6]
```diff
@@ -38,6 +38,7 @@ queue_rules:
               - "status-success=ci/centos/upgrade-tests-cephfs"
               - "status-success=ci/centos/upgrade-tests-rbd"
               - "status-success=DCO"
+              - "status-success=DCO"
```
Comment (Madhu-1) [inline .mergify.yml]: why this extra?

File: .mergify.yml [hunk 7]
```diff
@@ -53,6 +54,9 @@ queue_rules:
               - "status-success=ci/centos/mini-e2e/k8s-1.21"
               - "status-success=ci/centos/mini-e2e/k8s-1.22"
               - "status-success=ci/centos/mini-e2e/k8s-1.23"
+              - "status-success=ci/centos/k8s-e2e-external-storage/1.21"
+              - "status-success=ci/centos/k8s-e2e-external-storage/1.22"
+              - "status-success=ci/centos/k8s-e2e-external-storage/1.23"
```
Comment (Madhu-1) [inline .mergify.yml]: are we running these tests in release-3.4 branch?

**Review by Madhu-1 — CHANGES_REQUESTED** [review]

File: .mergify.yml (hunk 7 above)
Comment (Madhu-1) [reply .mergify.yml]: @humblec this is marked as resolved but not addressed yet?

**Comment by humblec** [comment]
@Madhu-1 can you please approve this PR, so that backport in doc PR works?

**Comment by humblec** [comment]
@Rakshith-R can you also revisit this PR please?

**Comment by Madhu-1** [comment]
> @Madhu-1 can you please approve this PR, so that backport in doc PR works?

https://github.com/ceph/ceph-csi/pull/2772/files#r783640380 is not addressed yet?

File: .mergify.yml (hunk 7 above)
Comment (humblec) [reply .mergify.yml]: @Madhu-1 done. missed to push last time.

File: .mergify.yml [hunk 8]
```diff
@@ -175,6 +175,47 @@ pull_request_rules:
...
+      - "approved-reviews-by=@ceph/ceph-csi-contributors"
+      - "approved-reviews-by=@ceph/ceph-csi-maintainers"
+      - "status-success=codespell"
+      - "status-success=multi-arch-build"
+      - "status-success=go-test"
+      - "status-success=commitlint"
+      - "status-success=golangci-lint"
+      - "status-success=mod-check"
+      - "status-success=lint-extras"
+      - "#changes-requested-reviews-by=0"
```
Comment (Madhu-1) [inline .mergify.yml]: this should have been added at line 193 to keep the same checks in the group

**Review by Madhu-1 — DISMISSED** [review]

File: .mergify.yml [hunk 9]
```diff
@@ -175,6 +175,47 @@ pull_request_rules:
...
+      - "status-success=ci/centos/k8s-e2e-external-storage/1.21"
+      - "status-success=ci/centos/k8s-e2e-external-storage/1.22"
+      - "status-success=ci/centos/k8s-e2e-external-storage/1.23"
+      - "status-success=ci/centos/upgrade-tests-cephfs"
+      - "status-success=ci/centos/upgrade-tests-rbd"
+      - "status-success=DCO"
+    actions:
+      queue:
+        name: default
+      dismiss_reviews: {}
```
Comment (Rakshith-R) [inline .mergify.yml]:
```suggestion
```
remove this line.
refer https://github.com/ceph/ceph-csi/pull/2764

**Review by Rakshith-R — DISMISSED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]

**Comment by humblec** [comment]
Merging this manually..

**Pull request merged**